### PR TITLE
userguide: add DHCP EVE log information v1

### PR DESCRIPTION
**Pull request by jmtaylor90**
Signed-off-by: jason taylor <jtfas90@gmail.com>

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
-
-
-

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: doc/userguide/output/eve/eve-json-format.rst [hunk 1]
```diff
@@ -2461,3 +2461,75 @@ Example of QUIC logging with a CYU hash:
...
+* "dns_servers": IP address(es) of servers the client will use for DNS queries
+
+Examples
+~~~~~~~~
+
+Example of DHCP log entry (default logging level):
+
+::
+
+
```
Comment (jufajardini) [inline doc/userguide/output/eve/eve-json-format.rst:2500]: nit: no need of those two extra lines

File: doc/userguide/output/eve/eve-json-format.rst [hunk 2]
```diff
@@ -2461,3 +2461,75 @@ Example of QUIC logging with a CYU hash:
...
+    "dhcp_type":"ack",
+    "renewal_time":21600,
+    "client_id":"54:ee:75:51:e0:66"
+  }
+
+
+Example of DHCP log entry (extended logging enabled):
+
+::
+
```
Comment (jufajardini) [inline doc/userguide/output/eve/eve-json-format.rst:2516]: nit: unneeded extra line.

File: doc/userguide/output/eve/eve-json-format.rst [hunk 3]
```diff
@@ -2461,3 +2461,75 @@ Example of QUIC logging with a CYU hash:
         }
     ]
   }
+
+Event type: DHCP
+-----------------
+
+The default logging level only logs DHCP reply types which contains the
```
Comment (jufajardini) [inline doc/userguide/output/eve/eve-json-format.rst:2468]: nit: s/contains/contain

File: doc/userguide/output/eve/eve-json-format.rst [hunk 4]
```diff
@@ -2461,3 +2461,75 @@ Example of QUIC logging with a CYU hash:
...
+MAC address and assigned IP of requesting clients.
+
+Fields
+~~~~~~
+
+* "type": DHCP message type
```
Comment (jufajardini) [inline doc/userguide/output/eve/eve-json-format.rst:2474]: Maybe remove the `DHCP` bit here, and add the possible types (`request`, `reply` or `<unknown`>), to try to reduce confusion of having two different fields with the same explanation?

**Review by jufajardini — CHANGES_REQUESTED** [review]
Very welcome addition, imo, thanks!

I think we can improve it a bit, please see my inline comments. :)

File: doc/userguide/output/eve/eve-json-format.rst (hunk 2 above)
Comment (jmtaylor90) [reply doc/userguide/output/eve/eve-json-format.rst:2516]: which line number is would you like removed?

Comment (jufajardini) [reply doc/userguide/output/eve/eve-json-format.rst:2516]: GH isn't great with line highlighting, sometimes. Sorry x_x
2516 works :)

**Comment by jmtaylor90** [comment]
continued in https://github.com/OISF/suricata/pull/8528

**Pull request closed without merging**